overlord/ifacestate/ifacemgr_test.go

```diff
@@ -93,13 +101,31 @@ func (s *interfaceManagerSuite) TearDownTest(c *C) {
 
 func (s *interfaceManagerSuite) manager(c *C) *ifacestate.InterfaceManager {
 	if s.privateMgr == nil {
-		mgr, err := ifacestate.Manager(s.state, s.extraIfaces)
+		mgr, err := ifacestate.Manager(s.state, s.hookManager(c), s.extraIfaces)
 		c.Assert(err, IsNil)
 		s.privateMgr = mgr
 	}
 	return s.privateMgr
 }
 
+func (s *interfaceManagerSuite) hookManager(c *C) *hookstate.HookManager {
+	if s.privateHookMgr == nil {
+		mgr, err := hookstate.Manager(s.state)
+		c.Assert(err, IsNil)
+		s.privateHookMgr = mgr
+	}
+	return s.privateHookMgr
+}
+
+func (s *interfaceManagerSuite) settle(c *C) {
+	for i := 0; i < 50; i++ {
```

Comment:
I really dislike this settle thing. We should have predictable and deterministic behaviour on how many times we need to call it and build some safeguards that check that after X iterations we're still changing the state so we're giving up.

Still, don't block on this. It's a general comment.